src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1987,6 +1987,9 @@ private function registerMailerConfiguration(array $config, ContainerBuilder $co
         $container->getDefinition('mailer.transports')->setArgument(0, $transports);
         $container->getDefinition('mailer.default_transport')->setArgument(0, current($transports));
 
+        $container->removeDefinition('mailer.logger_message_listener');
```

Comment:
why removing a definition and adding a deprecated alias following it ? Why not creating the right deprecated alias from the beginning ?

Reply:
The `mailer.message_logger_listener` service is not always defined (only when the profiler is enabled).

Reply:
but here, you still remove the `mailer.logger_message_listener` definition all the time to replace it by an alias. So I don't see how this matters.